### DISCUSSION TITLE
Diarization not working on some Nova-2 models.

### DISCUSSION THREAD

**Discussion by AbbasPlusPlus** [deepgram/community#939] [General help] [answered]
I've been using nova-2-medical and diarization was previously working but it has stopped. It also doesn't work on several of the other nova-2 models. I've changed to nova-2-general and it seems to work okay on there for now. I've seen other discussions which recommend changing models but that's not really a long term solution.

**Comment by jpvajda** [comment]
@AbbasPlusPlus  can you provide a request ID so we can dig deeper? Also sharing with us what API request you are attempting to make will help.

**Reply by AbbasPlusPlus** [reply]
Hey, by request ID, do you mean from the actual API call? Since I posted this, diarization has started working again for uploaded files but it's quite inconsistent. For the same file, sometimes it diarizes, other times it doesn't at all. I've also not been able to diarize real time SST at all. 

My code is almost exactly the same as the [Nextjs live transcription example](https://github.com/deepgram-starters/nextjs-live-transcription/blob/main/app/context/DeepgramContextProvider.tsx).

Here are the options I'm using specifically:

```
void connectToDeepgram({
        model: DEEPGRAM_MODEL,
        smart_format: true,
        language: DEEPGRAM_LANGUAGE,
        filler_words: true,
        diarize: true,
        punctuate: true,
        dictation: true,
      });
```


      
Here is the connectToDeepgram function in my provider
      
```
const connectToDeepgram = debounce(
    async (options: LiveSchema, endpoint?: string): Promise<void> => {
      const key = await getApiKey();
      const deepgram = createClient(key);

      const conn = deepgram.listen.live(options, endpoint);

      conn.addListener(LiveTranscriptionEvents.Open, () => {
        setConnectionState(SOCKET_STATES.open);
      });

      conn.addListener(LiveTranscriptionEvents.Close, () => {
        setConnectionState(SOCKET_STATES.closed);
      });

      setConnection(conn);
    },
    300,
  ) as (options: LiveSchema, endpoint?: string) => Promise<void>;
```

**Comment by jpvajda** [comment, marked as answer]
@AbbasPlusPlus That correct, all API requests will have a request ID field, if you have a some where diarization is performing poorly we can take a deeper look into our logs to see if there is any helpful info you can provide.

Also our Diarization feature has been having issues lately and it's been reported by a lot of users, So it's something our engineering teams will need to investigate more. 

You can also try using our Mulitchannel feature with Diarization for better performance, [this Doc](https://developers.deepgram.com/docs/multichannel-vs-diarization#combining-multichannel-with-diarization) can you give you some more information on combining them together.

```json
{
    "metadata": {
        "transaction_key": "deprecated",
        "request_id": "cb8b73e9-2c5c-42f9-a5e1-e613fc340fc1", <--- ID HERE
        "sha256": "958f93c5bc133bf5a1ad009f71cb0319af7a4f24544063d2f1e712fb4a16913e",
        "created": "2024-10-18T19:33:33.521Z",
        "duration": 13.78,
        "channels": 1,
        "models": [
            "1abfe86b-e047-4eed-858a-35e5625b41ee"
        ],
        "model_info": {
            "1abfe86b-e047-4eed-858a-35e5625b41ee": {
                "name": "2-general-nova",
                "version": "2024-01-06.5664",
                "arch": "nova-2"
            }
        }
    },
```
